web/modules/custom/jcc_newslinks/jcc_newslinks.module

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/**
+ * Implements hook_cron().
+ *
+ * Automatically unpromote NewsLinks after the 4th ranking that are older than 2 weeks.
+ */
+function jcc_newslinks_cron() {
+  $oldTimeValue = time() - (14 * 24 * 60 * 60); // timestamp for 14 days previous
+  $query = \Drupal::entityQuery('node')
+    ->condition('type', 'news')
+    ->condition('field_news_type.entity:taxonomy_term.name', 'NewsLink')
+    ->condition('promote', true) // items that only are promoted
+    ->condition('created', $oldTimeValue, '<') // items older than 14 days ago
+    ->condition('field_sorting_weight', -7, '>'); // items older than 4th ranking
```

Comment:
@nelsonwong-jcc field_sorting_weight may not always be -7.  The values change when they drag the items to sort.

Before this query, make another query to find out the $fourth_sorting_weight;

Roughly, it would look like this:
  $query = \Drupal::entityQuery('node')
    ->condition('type', 'news')
    ->condition('field_news_type.entity:taxonomy_term.name', 'NewsLink')
    ->condition('promote', true) // items that only are promoted
    ->condition('created', $oldTimeValue, '<') // items older than 14 days ago
    ->range(3, 1) // get the 4th item; indexed at 0.
    ->sort('field_sorting_weight', 'ASC');

// Get the nid from the query

// Then load the node
$node_storage = \Drupal::entityTypeManager()->getStorage('node');
$node = $node_storage->load($nid);

$fourth_sorting_weight = $node->field_sorting_weight->getString();  // if this doesn't work, try $node->get('field_sorting_weight')->getValue()

Reply:
@ivanbueno-jcc Okay, I got ya.

Reply:
@ivanbueno-jcc I've got the nid but the $node value is coming up null in the code below even though $nid is 9887.  Not sure why. As a result, the getString and getValue's are pulling in null and tossing an error too.  Any ideas?
` $node_storage = \Drupal::entityTypeManager()->getStorage('node');
  $node = $node_storage->load($nid);`

Reply:
@ivanbueno-jcc Doh.  Okay, the $nid is an array.  Guess I have to get the value out of it.